AEPAssurance/Source/AssuranceEvent.swift

```diff
@@ -50,7 +50,20 @@ struct AssuranceEvent: Codable {
     static func from(jsonData: Data) -> AssuranceEvent? {
         let decoder = JSONDecoder()
         decoder.dateDecodingStrategy = .millisecondsSince1970
-        guard var event = try? decoder.decode(AssuranceEvent.self, from: jsonData) else {
+        var event = try? decoder.decode(AssuranceEvent.self, from: jsonData)
+        if event == nil {
+            // If it fails, it must be an event with an encoded payload
+            var decodedMessage: [String: AnyCodable]? = nil
+            decodedMessage = try? decoder.decode([String: AnyCodable].self, from: jsonData)
+            if let payloadData = decodedMessage?["payload"]?.stringValue?.data(using: .utf8)  {
```

Comment:
Finally found the docs! Looking at the definitions of the [Control events](https://wiki.corp.adobe.com/display/adms/Assurance+SDK+-+Griffon+events#AssuranceSDKGriffonevents-EventType:Control) the `payload` is always a dictionary. Based on our earlier discussions it indeed seems to be an issue with the chunked event that the service is sending. IMO even though this works on Android as is - we should fix this on the service to avoid iOS doing this logic. Let us discuss with Adam on this when he is back.